packages/frontend-2/lib/settings/helpers/constants.ts

```diff
@@ -0,0 +1,178 @@
+}
+
+export const WorkspaceUserUpdateConfig: Record<
+  WorkspaceUserUpdateActionTypes,
+  WorkspaceUserUpdateActionConfig
+> = {
+  [WorkspaceUserUpdateActionTypes.MakeAdmin]: {
+    menu: {
+      title: 'Make admin...',
+      show: ({
+        isActiveUserWorkspaceAdmin = false,
+        isActiveUserTargetUser,
+        targetUserCurrentRole
+      }) =>
+        isActiveUserWorkspaceAdmin &&
+        !isActiveUserTargetUser &&
+        targetUserCurrentRole !== Roles.Workspace.Admin &&
+        targetUserCurrentRole !== Roles.Workspace.Guest
+    },
+    dialog: {
+      title: 'Make an admin',
+      mainMessage: (seatType) =>
+        seatType === WorkspaceSeatType.Editor
+          ? 'They will become project owner for all existing and new workspace projects.'
+          : 'They will be given an editor seat and become project owner for all existing and new workspace projects.',
+      showRoleInfo: true,
+      buttonText: 'Make an admin',
+      seatCountMessage: true
+    }
+  },
+  [WorkspaceUserUpdateActionTypes.RemoveAdmin]: {
+    menu: {
+      title: 'Remove as admin...',
+      show: ({
+        isActiveUserWorkspaceAdmin = false,
+        isActiveUserTargetUser,
+        targetUserCurrentRole
+      }) =>
+        isActiveUserWorkspaceAdmin &&
+        !isActiveUserTargetUser &&
+        targetUserCurrentRole === Roles.Workspace.Admin
+    },
+    dialog: {
+      title: 'Remove as admin',
+      mainMessage: 'They will lose admin privileges and become a member.',
+      showRoleInfo: false,
+      buttonText: 'Remove as admin',
+      seatCountMessage: false
+    }
+  },
+  [WorkspaceUserUpdateActionTypes.MakeGuest]: {
+    menu: {
+      title: 'Make guest...',
+      show: ({
+        isActiveUserWorkspaceAdmin = false,
+        isActiveUserTargetUser,
+        targetUserCurrentRole
+      }) =>
+        isActiveUserWorkspaceAdmin &&
+        !isActiveUserTargetUser &&
+        targetUserCurrentRole !== Roles.Workspace.Guest
+    },
+    dialog: {
+      title: 'Make a guest',
+      mainMessage: 'They will lose access to all existing workspace projects.',
+      showRoleInfo: true,
+      buttonText: 'Make a guest'
+    }
+  },
+  [WorkspaceUserUpdateActionTypes.MakeMember]: {
+    menu: {
+      title: 'Make member...',
+      show: ({
+        isActiveUserWorkspaceAdmin = false,
+        isActiveUserTargetUser,
+        targetUserCurrentRole
+      }) =>
+        isActiveUserWorkspaceAdmin &&
+        !isActiveUserTargetUser &&
+        targetUserCurrentRole === Roles.Workspace.Guest
+    },
+    dialog: {
+      title: 'Make member',
+      mainMessage: (seatType) =>
+        seatType === WorkspaceSeatType.Editor
+          ? 'They will be given a viewer seat and lose project ownership.'
+          : 'They will be given a viewer seat.',
+      showRoleInfo: true,
+      buttonText: 'Make a member'
+    }
+  },
+  [WorkspaceUserUpdateActionTypes.UpgradeEditor]: {
```

Comment:
I do wonder if these should just be components that update upgrade dialog component, it's just my personal preference of composition over configuration, because now if you want one modal that is slightly different (which we tend to do haha) you have to update the config to fit your new needs. Let's leave for now, and we can take a look once we have the new designs for the confirmation modal to compare